UPDATING.md

```diff
@@ -0,0 +1,127 @@
+```
+
+Adding the necessary code to stay on top of cancellation, you will need something like this:
+
+```go
+ctx, cancelFunc := context.WithTimeout(context.Background(), 10 * time.Second)
+defer cancelFunc()
+```
+
+This way, the context can be cancelled at the close of the function.
+
+### Client Options now has a Functional Interface
+
+Before version 1.2.0, you instantiated a client using a `ClientOptions` instance created with the `NewClientOptions` function:
+
+```go
+client, err := ably.NewRealtime(ably.NewClientOptions("xxx:xxx"))
+```
+
+**Starting with version 1.2.0**, you must use the new functional interface:
+
+```go
+client, err := ably.NewRealtime(ably.WithKey("xxx:xxx"))
+```
+
+For a full list of client options, see all functions prefixed by `With` that return a `ClientOption` function
+[in the API reference](https://pkg.go.dev/github.com/ably/ably-go/ably#ClientOption).
+
+### Subscription now uses Message Handlers
+
+Before version 1.2.0, you subscribed to receive all messages from a Go channel like this:
+
+```go
+sub, _ := channel.Subscribe()
+for msg := range sub.MessageChannel() {
+    fmt.Println("Received message: ", msg)
+}
+```
+
+**Starting with version 1.2.0**, you must supply a context and your own message handler function to the new [SubscribeAll](https://pkg.go.dev/github.com/ably/ably-go/ably#RealtimeChannel.SubscribeAll) method:
+
+```go
+unsubscribe, _ := channel.SubscribeAll(ctx, func(msg *ably.Message) {
+    fmt.Println("Received message: ", msg)
+})
+```
+
+The signature of the [Subscribe](https://pkg.go.dev/github.com/ably/ably-go/ably#RealtimeChannel.Subscribe) method has also changed. It now requires a [Context](https://pkg.go.dev/context#Context) as well as the channel name and your message handler.
+
+Both `Subscribe` and `SubscribeAll` are now blocking methods.
+
+### The `Publish` Method now Blocks
+
+Before version 1.2.0, you published messages to a channel by calling the `Publish` method and then waiting for the `Result`:
+
+```go
+result, _ := channel.Publish("EventName1", "EventData1")
+
+// block until the publish operation completes
+result.Wait()
+```
+
+**Starting with version 1.2.0**, you must supply a context, as this method is now blocking:
+
+```go
+// block until the publish operation completes or is cancelled
+channel.Publish(ctx, "EventName1", "EventData1")
+```
+
+### Querying History
+
+Before version 1.2.0, you queried history as shown below, receiving a `PaginatedResult` from the `History` function:
+
+```go
+page, err := channel.History(nil)
+for ; err == nil; page, err = page.Next() {
+    for _, message := range page.Messages() {
+        fmt.Println("Message from History: ", message)
+    }
+}
+if err != nil {
+    panic(err)
+}
+```
+
+**Starting with version 1.2.0**, you must use the blocking [Pages](https://pkg.go.dev/github.com/ably/ably-go/ably#HistoryRequest.Pages) method on the returned `HistoryRequest` instance:
```

Comment:
It was also blocking before.

It may be worth mentioning that History now takes functional options, instead of a Params struct.